src/js/tools/pen.js

```diff
@@ -66,25 +66,31 @@ define(function (require, exports, module) {
                     return descriptor.playObject(optionsObj);
                 });
 
-        if (this.vectorMode) {
-            var selectVectorMask = descriptor.playObject(vectorMaskLib.activateVectorMaskEditing());
-            return Promise.join(setPromise, selectVectorMask);
+        var deleteFn = function (event) {
+            event.stopPropagation();
+
+            return PS.performMenuCommand(_CLEAR_PATH)
+                .catch(function () {
+                    // Silence the errors here
+                });
+        };
+
+        // Disable target path suppression
+        var backspacePromise = this.transfer(shortcuts.addShortcut,
+                OS.eventKeyCode.BACKSPACE, {}, deleteFn, "penBackspace", true),
+            deletePromise = this.transfer(shortcuts.addShortcut,
+                OS.eventKeyCode.DELETE, {}, deleteFn, "penDelete", true),
+            disableSuppressionPromise = UI.setSuppressTargetPaths(false);
+
+        if (vectorMode) {
+            var selectVectorMask = UI.setSuppressTargetPaths(false)
+                .then(function () {
+                    return descriptor.playObject(vectorMaskLib.activateVectorMaskEditing());
+                });
+
+            return Promise.join(setPromise, selectVectorMask,
```

Comment:
(You can do this by having an if-then-else statement that initializes a variable to a promise returned by an adapter call in one branch and to `Promise.resolve()` in the other branch.)